NWN.Anvil/src/main/Services/API/Creature/InitiativeModifierService.cs

```diff
@@ -0,0 +1,118 @@
+using System;
+using Anvil.API;
+using Anvil.Internal;
+using NWN.Native.API;
+using Feat = Anvil.API.Feat;
+
+namespace Anvil.Services
+{
+  [ServiceBinding(typeof(InitiativeModifierService))]
+  [ServiceBindingOptions(InternalBindingPriority.API, Lazy = true)]
+  public sealed unsafe class InitiativeModifierService
+  {
+    private readonly FunctionHook<InitiativeModifier> initiativeModiferHook;
+
+    public InitiativeModifierService(HookService hookService)
+    {
+      initiativeModiferHook = hookService.RequestHook<InitiativeModifier>(OnResolveInitiative, FunctionsLinux._ZN12CNWSCreature17ResolveInitiativeEv, HookOrder.Late);
+    }
+
+    private delegate void InitiativeModifier(void* pObject);
+
+    /// <summary>
+    /// Clears any modifier that is set for the creature's initiative.<br/>
+    /// </summary>
+    public void ClearInitiativeModifier(NwCreature creature)
+    {
+      InternalVariables.InitiativeMod(creature).Delete();
+    }
+
+    /// <summary>
+    /// Gets the modifier that is set for the creature's initiative.<br/>
+    /// </summary>
+    public int? GetInitiativeModifier(NwCreature creature)
+    {
+      InternalVariableInt initiativeModifier = InternalVariables.InitiativeMod(creature);
+      if (initiativeModifier.HasValue)
+      {
+        return InternalVariables.InitiativeMod(creature);
+      }
+
+      return null;
+    }
+
+    /// <summary>
+    /// Sets the modifier that is set for the creature's initiative.<br/>
+    /// </summary>
+    public void SetInitiativeModifier(NwCreature creature, int mod)
+    {
+      InternalVariables.InitiativeMod(creature).Value = mod;
+    }
+
+    private void OnResolveInitiative(void* pCreature)
+    {
+      NwCreature? creature = CNWSCreature.FromPointer(pCreature).ToNwObject<NwCreature>();
+      if (creature == null)
+      {
+        initiativeModiferHook.CallOriginal(pCreature);
+        return;
+      }
+
+      InternalVariableInt initMod = InternalVariables.InitiativeMod(creature);
+      if (initMod.HasNothing)
+      {
+        initiativeModiferHook.CallOriginal(pCreature);
+        return;
+      }
+
+      CNWSCreature cCreature = creature.Creature;
+
+      if (cCreature.m_bInitiativeExpired == 1)
```

Comment:
Prefer usage of the `ToBool()` extension instead of the int comparison.
```suggestion
      if (cCreature.m_bInitiativeExpired.ToBool())
```